src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -451,6 +464,13 @@ bool CoreWorkerMemoryStore::Contains(const ObjectID &object_id, bool *in_plasma)
   return false;
 }
 
+void CoreWorkerMemoryStore::OnErase(std::shared_ptr<RayObject> obj) {
+  if (obj->IsException() && !obj->IsInPlasmaError() && !obj->WasAccessed() &&
```

Comment:
In OOM cases, I think the raylet is the one that sets the error, by putting the error in plasma. So this will not throw the error for those cases. We should centralize the error handling to the in-memory value. Can you add a TODO to return errors from the raylet to the owner?